src/adapters/DICOMMicroscopyViewer/MeasurementReport.js

```diff
@@ -0,0 +1,182 @@
+import { Normalizer } from '../../normalizers.js';
+import { DicomMetaDictionary } from '../../DicomMetaDictionary.js';
+import { StructuredReport } from '../../derivations.js';
+import TID1500MeasurementReport from '../../utilities/TID1500/TID1500MeasurementReport.js';
+import TID1501MeasurementGroup from '../../utilities/TID1500/TID1501MeasurementGroup.js';
+
+import { toArray, codeMeaningEquals } from '../helpers.js';
+
+function getTID300ContentItem(tool, graphicType, ReferencedSOPSequence, toolClass) {
+  const args = toolClass.getTID300RepresentationArguments(tool);
+  args.ReferencedSOPSequence = ReferencedSOPSequence;
+  args.use3DSpatialCoordinates = true;
+  return new toolClass.TID300Representation(args);
+}
+
+function getMeasurementGroup(graphicType, measurements, ReferencedSOPSequence) {
+  const toolClass = MeasurementReport.MICROSCOPY_TOOL_CLASSES_BY_TOOL_TYPE[graphicType];
+
+  // Loop through the array of tool instances
+  // for this tool
+  const Measurements = measurements.map(tool => {
+    return getTID300ContentItem(tool, graphicType, ReferencedSOPSequence, toolClass);
+  });
+
+  return new TID1501MeasurementGroup(Measurements);
+}
+
+export default class MeasurementReport {
+  constructor() {
+  }
+
+  static generateReport(rois, metadataProvider, options) {
+    // Input is all ROIS returned via viewer.getALLROIs()
+    // let report = MeasurementReport.generateReport(viewer.getAllROIs());
+
+    // Sort and split into arrays by scoord.graphicType
+    const measurementsByGraphicType = {};
+    rois.forEach(roi => {
+      const graphicType = roi.scoord.graphicType;
+
+      if (!measurementsByGraphicType[graphicType]) {
+        measurementsByGraphicType[graphicType] = [];
+      }
+
+      measurementsByGraphicType[graphicType].push(roi.scoord);
+    });
+
+    // For each measurement, get the utility arguments using the adapter, and create TID300 Measurement
+    // Group these TID300 Measurements into a TID1501 Measurement Group (for each graphicType)
+    // Use TID1500MeasurementReport utility to create a single report from the created groups
+    // return report;
+
+    // TODO: Find a way to get this from the measurement itself
+    const ReferencedSOPSequence = {
```

Comment:
You can remove this since it isn't used for SCOORD3D.

Reply:
done. Thank you!